pol-core/pol/module/osmod.h

```diff
@@ -118,6 +118,7 @@ class OSExecutorModule : public Bscript::TmplExecutorModule<OSExecutorModule, Co
 
   [[nodiscard]] Bscript::BObjectImp* mf_PerformanceMeasure();
   [[nodiscard]] Bscript::BObjectImp* mf_LoadExportedScript();
+  [[nodiscard]] Bscript::BObjectImp* mf_GetEnvironmentVariable();
```

Comment:
This seems like a security risk.  Can we have a config file that contains a list of environment variables names that are allowed to be queried?

Reply:
Mmmh in which way is getting an environment variable a security risk?
For example in my old shard we ran two servers where special files where needed to be able to distinguish between them. Via an environment variable this would be way easier.
The same is true for SQL database password, instead of a cfg file with the password which could be by accident checked in an environment variable on the server is much safer.

Reply:
The SQL database password provides the best example.  If any script in any package can read any environment variable, then if a database password is present in the environment this presents a security risk, in that any time you install any package you have to be aware that it could access that database password.

Reply:
Mmmh funny attack vector.

Seems we have fundamental different opinion how scripts are shared and what intellectual abilities the shard admins have.
Added a pol.cfg setting..

Reply:
This should be solved now with the setting to whitelist environment variables added by @KevinEady

Reply:
@eric-swanson environment variables are NOT a safe place to store any secrets AT ALL, at ANY GIVEN TIME. Therefore, in my view, the argument to begin with is invalid. On the other hand, it enables you to gain incredible flexibility without too much investment in your code.

Reply:
Hi Karol, thanks for your comments. I believe we found a good compromise. I'm marking this as resolved now.